privacyidea/lib/event.py

```diff
@@ -56,12 +58,49 @@ def __call__(self, func):
         def event_wrapper(*args, **kwds):
             # here we have to evaluate the event configuration from the
             # DB table eventhandler and based on the self.eventname etc...
-            # TODO: do Pre-Event Handling
+            # do Pre-Event Handling
+            e_handles = self.g.event_config.get_handled_events(self.eventname, position="pre")
+            for e_handler_def in e_handles:
```

Comment:
What do we do if a pre event handler fails, i.e. raises an Exception?

Reply:
The event handler itself decides if it will raise an exception and thus terminate the request.

Reply:
:+1: